sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetUtils.scala

```diff
@@ -195,7 +201,12 @@ object ParquetUtils {
       case (_, i) =>
         throw new SparkException("Unexpected parquet type name: " + primitiveTypeNames(i))
     }
-    converter.currentRecord
+
+    if (aggregation.groupByColumns.length > 0) {
+      new JoinedRow(partitionValues, converter.currentRecord)
```

Comment:
just curious how we make sure the `partitionValues` is always consistent with `aggregation.groupByColumns`? in terms of ordering.

Reply:
I actually forgot to considering ordering. Fixed. Thanks for pointing this out!